honeybee_radiance/modifierset.py

```diff
@@ -0,0 +1,702 @@
+"""Radiance modifier set."""
+from __future__ import division
+
+from honeybee_radiance.primitive import Primitive
+from honeybee_radiance.lib.modifiers import (
+    generic_exterior_wall, generic_interior_wall,
+    generic_roof, generic_ceiling,
+    generic_floor, air_wall,
+    generic_exterior_glass, generic_interior_glass
+)
+from honeybee._lockable import lockable
+import honeybee.typing as typing
+
+
+_default_modifiers = {
+    '_BaseSet': {'exterior': None, 'interior': None},
+    'WallSet': {
+        'exterior': generic_exterior_wall,
+        'interior': generic_interior_wall
+    },
+    'FloorSet': {
+        'exterior': generic_floor,
+        'interior': generic_floor
+    },
+    'RoofCeilingSet': {
+        'exterior': generic_roof,
+        'interior': generic_ceiling
+    },
+    'ApertureSet': {
+        'exterior': generic_exterior_glass,
+        'interior': generic_interior_glass,
+        'skylight': generic_exterior_glass,
+        'glass_door': generic_exterior_glass
+    },
+    'DoorSet': {
+        'exterior': generic_exterior_wall,
+        'interior': generic_interior_wall,
+        'overhead': generic_interior_wall
+    }
+}
+
+
+@lockable
+class _BaseSet(object):
+    """Base class for the sets assigned to Faces (WallSet, FloorSet, RoofCeilingSet)."""
+
+    __slots__ = ('_exterior_modifier', '_interior_modifier', '_locked')
+
+    def __init__(self, exterior_modifier=None, interior_modifier=None):
+        """Initialize set.
+
+        Args:
+            exterior_modifier: A radiance modifier object for faces with an
+                Outdoors boundary condition.
+            interior_modifier: A radiance modifier object for faces with a boundary
+                condition other than Outdoors.
+        """
+        self._locked = False  # unlocked by default
+        self.exterior_modifier = exterior_modifier
+        self.interior_modifier = interior_modifier
+
+    @property
+    def exterior_modifier(self):
+        """Get or set the modifier for exterior Faces."""
+        if self._exterior_modifier is None:
+            return _default_modifiers[self.__class__.__name__]['exterior']
+        return self._exterior_modifier
+
+    @exterior_modifier.setter
+    def exterior_modifier(self, value):
+        self._exterior_modifier = self._validate_modifier(value)
+
+    @property
+    def interior_modifier(self):
+        """Get or set the modifier for interior Faces."""
+        if self._interior_modifier is None:
+            return _default_modifiers[self.__class__.__name__]['interior']
+        return self._interior_modifier
+
+    @interior_modifier.setter
+    def interior_modifier(self, value):
+        self._interior_modifier = self._validate_modifier(value)
+
+    @property
+    def modifiers(self):
+        """List of all modifiers contained within the set."""
+        return [getattr(self, attr[1:]) for attr in self._slots]
+
+    @property
+    def modified_modifiers(self):
```

Comment:
Hm. I realize that I may not have thought through all of the implications of this naming choice.  I don't really have a better solution to offer at this time but, if you come up with a better convention that `modified`, I am happy to implement it on the honeybee-energy side as well.

Reply:
I didn't want to suggest a major change in the middle of your current development and I as I said I tried to keep the methods the exact same name. I personally think we can use `set` instead of `modified`. These are the modifiers / constructions that are __set__ by user. In this case it will be `set_modifiers` or `modifiers_set_by_user`.

I should also say that it is fine to have it as is. One can read the docstring and that should tell them what this is.